### PR TITLE
Added smoke test for Axes.imshow to test_datetime.py

### PR DESCRIPTION
**Pull request by ConstableCatnip**
## PR summary
This PR adds a smoke test for `Axes.imshow` in lib/matplotlib/tests/test_datetime.py and addresses task `Axes.imshow` in issue https://github.com/matplotlib/matplotlib/issues/26864. Due to the behavior of `Axes.imshow` with the `mpl.rcParams["date.converter"] = "concise"`, I included year, month, day, and time axis scale plots.
Below is the image of the generated plot:

image

## PR checklist
- [x] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [NA ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [NA] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [NA] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: lib/matplotlib/tests/test_datetime.py [hunk 1]
```diff
@@ -530,11 +530,45 @@ def test_hlines(self):
...
+        a = np.diag(range(5))
+        fig, axes = plt.subplots(nrows=2, ncols=2)
+
+        ax = axes[0, 0]
+        dt_start = datetime.datetime(1980, 4, 15)
+        dt_end = datetime.datetime(2010, 11, 11)
+        extent = (dt_start, dt_end, dt_start, dt_end)
+        ax.imshow(a, extent=extent)
+        for label in ax.get_xticklabels():
+            label.set_rotation(90)
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_datetime.py:545]:
```suggestion
        ax.tick_params(axis='x', labelrotation=90)
```

File: lib/matplotlib/tests/test_datetime.py [hunk 2]
```diff
@@ -530,11 +530,45 @@ def test_hlines(self):
...
+
+        ax = axes[0, 1]
+        dt_start = datetime.datetime(2010, 4, 15)
+        dt_end = datetime.datetime(2010, 11, 11)
+        extent = (dt_start, dt_end, dt_start, dt_end)
+        ax.imshow(a, extent=extent)
+        for label in ax.get_xticklabels():
+            label.set_rotation(90)
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_datetime.py:553]:
```suggestion
        ax.tick_params(axis='x', labelrotation=90)
```

File: lib/matplotlib/tests/test_datetime.py [hunk 3]
```diff
@@ -530,11 +530,45 @@ def test_hlines(self):
...
+
+        ax = axes[1, 0]
+        dt_start = datetime.datetime(2010, 11, 1)
+        dt_end = datetime.datetime(2010, 11, 11)
+        extent = (dt_start, dt_end, dt_start, dt_end)
+        ax.imshow(a, extent=extent)
+        for label in ax.get_xticklabels():
+            label.set_rotation(90)
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_datetime.py:561]:
```suggestion
        ax.tick_params(axis='x', labelrotation=90)
```

File: lib/matplotlib/tests/test_datetime.py [hunk 4]
```diff
@@ -530,11 +530,45 @@ def test_hlines(self):
...
+
+        ax = axes[1, 1]
+        dt_start = datetime.datetime(2010, 11, 10)
+        dt_end = datetime.datetime(2010, 11, 11)
+        extent = (dt_start, dt_end, dt_start, dt_end)
+        ax.imshow(a, extent=extent)
+        for label in ax.get_xticklabels():
+            label.set_rotation(90)
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_datetime.py:569]:
```suggestion
        ax.tick_params(axis='x', labelrotation=90)
```

File: lib/matplotlib/tests/test_datetime.py [hunk 5]
```diff
@@ -530,11 +530,45 @@ def test_hlines(self):
...
                          xmax=0.65)
 
-    @pytest.mark.xfail(reason="Test for imshow not written yet")
     @mpl.style.context("default")
     def test_imshow(self):
-        fig, ax = plt.subplots()
-        ax.imshow(...)
+        mpl.rcParams["date.converter"] = "concise"
+        a = np.diag(range(5))
+        fig, axes = plt.subplots(nrows=2, ncols=2)
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_datetime.py:537]:
```suggestion
        fig, axes = plt.subplots(nrows=2, ncols=2, layout='compressed')
```

File: lib/matplotlib/tests/test_datetime.py [hunk 6]
```diff
@@ -530,11 +530,45 @@ def test_hlines(self):
...
+
+        ax = axes[1, 1]
+        dt_start = datetime.datetime(2010, 11, 10)
+        dt_end = datetime.datetime(2010, 11, 11)
+        extent = (dt_start, dt_end, dt_start, dt_end)
+        ax.imshow(a, extent=extent)
+        for label in ax.get_xticklabels():
+            label.set_rotation(90)
+
+        fig.tight_layout()
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_datetime.py:571]: Better to use the `layout` argument above.
```suggestion
```

**Review by QuLogic — COMMENTED** [review]
I'm not too sure we need all 4 scales, as that is testing the concise date formatter, not anything specific to `imshow`.

**Review by jklymak — APPROVED** [review]

**Review by QuLogic — APPROVED** [review]

**Pull request merged**